shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitParameters.java

```diff
@@ -51,13 +75,38 @@ public static ShortCircuitParameters load(PlatformConfig platformConfig) {
                         .setWithFeederResult(config.getBooleanProperty("with-feeder-result", DEFAULT_WITH_FEEDER_RESULT))
                         .setStudyType(config.getEnumProperty("study-type", StudyType.class, DEFAULT_STUDY_TYPE))
                         .setWithFortescueResult(config.getBooleanProperty("with-fortescue-result", DEFAULT_WITH_FORTESCUE_RESULT))
-                        .setMinVoltageDropProportionalThreshold(config.getDoubleProperty("min-voltage-drop-proportional-threshold", DEFAULT_MIN_VOLTAGE_DROP_PROPORTIONAL_THRESHOLD)));
+                        .setMinVoltageDropProportionalThreshold(config.getDoubleProperty("min-voltage-drop-proportional-threshold", DEFAULT_MIN_VOLTAGE_DROP_PROPORTIONAL_THRESHOLD))
+                        .setSubTransientCoefficient(config.getDoubleProperty("sub-transient-coefficient", DEFAULT_SUB_TRANSIENT_COEFFICIENT))
+                        .setWithLoads(config.getBooleanProperty("with-loads", DEFAULT_WITH_LOADS))
+                        .setWithShuntCompensators(config.getBooleanProperty("with-shunt-compensators", DEFAULT_WITH_SHUNT_COMPENSATORS))
+                        .setWithVSCConverterStations(config.getBooleanProperty("with-vsc-converter-stations", DEFAULT_WITH_VSC_CONVERTER_STATIONS))
+                        .setWithNeutralPosition(config.getBooleanProperty("with-neutral-position", DEFAULT_WITH_NEUTRAL_POSITION))
+                        .setInitialVoltageProfileMode(config.getEnumProperty("initial-voltage-profile-mode", InitialVoltageProfileMode.class, DEFAULT_INITIAL_VOLTAGE_PROFILE_MODE))
+                        .setVoltageRangeData(getVoltageRangeDataFromConfig(config, platformConfig)));
 
+        parameters.validate();
         parameters.readExtensions(platformConfig);
 
         return parameters;
     }
 
+    private static List<VoltageRangeData> getVoltageRangeDataFromConfig(ModuleConfig config, PlatformConfig platformConfig) {
+        Optional<Path> optionalVoltageRangeDataPath = config.getOptionalPathProperty("voltage-range-data");
+        if (optionalVoltageRangeDataPath.isPresent()) {
```

Comment:
Use optionalVoltageRangeDataPath.ifPresent(...) instead

Reply:
But it won't work with the return value no?